### ISSUE TITLE
No error for assignment to read-only property before definition

### ISSUE THREAD

**Issue by JelleZijlstra** [python/mypy#6734]
The following code:
```python
class X:
    def update(self) -> None:
        self.y = "something"

    @property
    def y(self) -> str:
        return "hello"
```
produces no errors (tried both 0.701 and master). It should fail on the assignment to `self.y`, which is a read-only property.

If the order of the two methods is reversed, an error is produced correctly.

Also, an error is produced correctly when I use `--new-semantic-analyzer`, so this will be fixed when the new semantic analyzer is turned on.

**Comment by ilevkivskyi** [comment]
Note that no error is reported even if there is a type mismatch (this is also fixed in the new analyzer).

**Comment by ilevkivskyi** [comment]
And also the error appears if the order is reversed.

**Comment by JukkaL** [comment]
Closing since this is fixed in the new semantic analyzer.
